Detectors/ITSMFT/ITS/macros/test/CheckLUtime.C

```diff
@@ -26,46 +27,58 @@ using namespace std;
 
 void CheckLUtime(std::string clusfile = "o2clus_its.root", std::string dictfile = "")
 {
-  using o2::itsmft::Cluster;
+  using o2::itsmft::ClusterPattern;
   using o2::itsmft::CompClusterExt;
   using o2::itsmft::LookUp;
+  using ROFRec = o2::itsmft::ROFRecord;
+
+  TStopwatch sw;
+  sw.Start();
 
   if (dictfile.empty()) {
     dictfile = o2::base::NameConf::getDictionaryFileName(o2::detectors::DetID::ITS, "", ".bin");
   }
+  std::ifstream file(dictfile.c_str());
+  if (file.good()) {
+    LOG(INFO) << "Running with dictionary: " << dictfile.c_str();
+  } else {
+    LOG(INFO) << "Running without dictionary !";
+  }
   LookUp finder(dictfile.c_str());
   ofstream time_output("time.txt");
 
   ofstream realtime, cputime;
   realtime.open("realtime.txt", std::ofstream::out | std::ofstream::app);
   cputime.open("cputime.txt", std::ofstream::out | std::ofstream::app);
 
-  TStopwatch timerLookUp;
-
   // Clusters
-  TFile* file1 = TFile::Open(clusfile.data());
-  TTree* clusTree = (TTree*)gFile->Get("o2sim");
-  std::vector<CompClusterExt>* clusArr = nullptr;
-  clusTree->SetBranchAddress("ITSClusterComp", &clusArr);
+  TFile* fileCl = TFile::Open(clusfile.data());
+  TTree* clusTree = (TTree*)fileCl->Get("o2sim");
   std::vector<unsigned char>* patternsPtr = nullptr;
   auto pattBranch = clusTree->GetBranch("ITSClusterPatt");
   if (pattBranch) {
     pattBranch->SetAddress(&patternsPtr);
   }
 
-  Int_t nevCl = clusTree->GetEntries(); // clusters in cont. readout may be grouped as few events per entry
-  int ievC = 0, ievH = 0;
+  // ROFrecords
+  std::vector<ROFRec> rofRecVec, *rofRecVecP = &rofRecVec;
+  clusTree->SetBranchAddress("ITSClustersROF", &rofRecVecP);
+  clusTree->GetEntry(0);
+  int nROFRec = (int)rofRecVec.size();
+
+  auto pattIdx = patternsPtr->cbegin();
+
+  int nClusters = 0;
+
+  for (int irof = 0; irof < nROFRec; irof++) {
+    const auto& rofRec = rofRecVec[irof];
+
+    rofRec.print();
 
-  for (ievC = 0; ievC < nevCl; ievC++) {
-    clusTree->GetEvent(ievC);
-    Int_t nc = clusArr->size();
-    printf("processing cluster event %d\n", ievC);
-    bool restart = false;
-    restart = (ievC == 0) ? true : false;
-    timerLookUp.Start(restart);
-    auto pattIdx = patternsPtr->cbegin();
-    for (int i = 0; i < nc; i++) {
-      CompClusterExt& c = (*clusArr)[i];
+    for (int icl = 0; icl < rofRec.getNEntries(); icl++) {
+      nClusters++;
+      int clEntry = rofRec.getFirstEntry() + icl; // entry of icl-th cluster of this ROF in the vector of clusters
+      // do we read MC data?
       auto rowSpan = *pattIdx++;
```

Comment:
This makes sense only for clusters created w/o a dictionary, i.e. ``cluster.getPatternID() == CompCluster::InvalidPatternID``, since most of clusters (unless they have groupID pattern)  will not have pattern stored.
Could you add a check and quit if the cluster has valid pattern?